src/Microsoft.AspNet.Security.OpenIDConnect/Notifications/AuthorizationCodeReceivedNotification.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved. See License.txt in the project root for license information.
+
+using Microsoft.AspNet.Http;
+using Microsoft.AspNet.Security.OpenIdConnect;
+using Microsoft.IdentityModel.Protocols;
+using System.Diagnostics.CodeAnalysis;
+using System.IdentityModel.Tokens;
+
+namespace Microsoft.AspNet.Security.Notifications
+{
+    /// <summary>
+    /// This Notification can be used to be informed when an 'AuthorizationCode' is received over the OpenIdConnect protocol.
+    /// </summary>
+    public class AuthorizationCodeReceivedNotification : BaseNotification<OpenIdConnectAuthenticationOptions>
+    {
+        /// <summary>
+        /// Creates a <see cref="AuthorizationCodeReceivedNotification"/>
+        /// </summary>
+        public AuthorizationCodeReceivedNotification(HttpContext context, OpenIdConnectAuthenticationOptions options) : base(context, options)
```

Comment:
Style: you should move `: base(...)` to a new line to make it consistent with the other providers:

https://github.com/aspnet/Security/blob/dev/src/Microsoft.AspNet.Security.Cookies/Notifications/CookieApplyRedirectContext.cs#L24